# Pneumococcal genome library and core genome multilocus sequence typing scheme

This repository contains the data and the analysis workflow for the manuscript "Development of core genome multilocus sequence typing and taxonomic barcoding schemes to delineate global pneumococcal population structure."

The repository is divided into 3 sections:  
1). Production - containing all the commands used to produce data for analyses.  
2). Analysis - containing all the commands used to analyse data.  
3). Figures - containing scripts (1 script per figure) used to plot the results.  

## Abstract
Background: Genomic epidemiology and surveillance are increasingly integral public health measures for understanding the distribution of pneumococcal genetic lineages associated with colonisation, disease, and antimicrobial resistance, and for assessing population-based changes due to pneumococcal vaccine implementation. The rapid accumulation of whole genome sequence data in recent years provides a strong incentive for the creation of accessible, high-quality genomic resources and genome-wide typing schemes for the scientific community.

Methods: We developed the Pneumococcal Genome Library (PGL), a curated dataset of 30,976 genomes derived from 115 published papers. We leveraged the diversity of this dataset to develop a core genome multilocus sequence typing (cgMLST) scheme comprising 1,222 highly conserved loci. Using multilevel single-linkage clustering, we stratified populations into hierarchical clusters based on allelic similarity thresholds, which we described with a taxonomic life identification number (LIN) barcoding system. The PGL and cgMLST barcoding schemes are publicly available within the PubMLST Streptococcus pneumoniae database (https://pubmlst.org/organisms/streptococcus-pneumoniae). 

Results: The PGL represents a diverse collection of carriage and disease pneumococci recovered between 1916-2018 from 82 countries, comprising 4,401 multilocus sequence types and 95 serotypes, including all 24 pneumococcal vaccine serotypes. Our LIN barcoding scheme clustered these pneumococci at 10 hierarchical classification levels, ranging from diverse, multinational lineages and sublineages down to within-host clonal populations. This allowed us to define 407 distinct pneumococcal lineages and 726 sublineages while simultaneously allowing the precise differentiation of populations from localised, high transmission settings. We then evaluated the global distribution and epidemiological characteristics of high prevalence, globally dispersed lineages.

Conclusions: We have produced a high-quality genomic resource and a fine-scale clustering approach to provide a unified, community-accessible framework for the analysis of pneumococcal populations, which supports surveillance and vaccine implementation efforts. 

